services/dedup/keydb/keydb.go

```diff
@@ -51,7 +49,7 @@ func NewKeyDB(conf *config.Config, stat stats.Stats, log logger.Logger) (*Dedup,
 	db.stats.getTimer = stat.NewTaggedStat("dedup_get_duration_seconds", stats.TimerType, stats.Tags{"mode": "keydb"})
 	db.stats.setTimer = stat.NewTaggedStat("dedup_set_duration_seconds", stats.TimerType, stats.Tags{"mode": "keydb"})
 
-	return &Dedup{keyDB: db}, nil
+	return db, nil
 }
 
 func (d *KeyDB) Get(keys []string) (map[string]bool, error) {
```

Comment:
doesn't need to be exported
```suggestion
func (d *keyDB) Get(keys []string) (map[string]bool, error) {
```